src/engine/Operation.h

```diff
@@ -211,8 +211,9 @@ class Operation {
   void checkAbortion() const;
   // Check if the abortion flag has been set and throws an exception if that's
   // the case. This will be called at strategic places on code that potentially
-  // can take a (too) long time.
-  void checkAbortion(const auto& detailSupplier) const;
+  // can take a (too) long time. The `detailSupplier` allows to pass a message
+  // to add to any potential exception that might be thrown.
```

Comment:
Also rework the first part of the comment:
You have found out that it is only a single read from memory with a perfectly predictable branch, so it can typically be also called in a relatively tight loop. The only caveat is that it prevents vectorization for numeric loops etc.

I think it is crucial to document this here s.t. people know how (in)-expensive it is.